Jean S Ariza

jesuarva opened the pull request:
# Original Front APP Trello: There are the latest commits.
- https://trello.com/c/gt45IoLd
- https://trello.com/c/XQpGyNkk

# New Repo for the Front-APP:
I was facing porblems deploying the original Repo, so, literally I copy the core files to a new folder, then I init a new github and local repo, installe dhte dependencies adn lastly deployed to Natlify -> Those acitons solved the issue deploying.
I am still confused to what happened with the first Repo
- https://github.com/jesuarva/lambda-debug

# Trello Boards:
- APP: https://trello.com/b/ntqeFPwr/lambda-notes-jean-s-ariza
- API: https://trello.com/b/6W18iHpl/lambda-notes-backend-jean-s-ariza

jesuarva committed:
ADD package.json && .gitignore

jesuarva committed:
ADD Trello URL to README.

jesuarva committed:
ADD Dependencies.

jesuarva committed:
ADD:

1. dotenv as a dev-dependencie
2. /secrets && .env to .gitignore

jesuarva committed:
DEFINE Starting files.

jesuarva committed:
REFACTOR RouterFactory.js

Adapt some class methods in order to accept dynamic params.

jesuarva committed:
ADD to package.json a script to launch Nodemon

jesuarva committed:
ADD Connection to DB (Temporarily a locally installed Mongod)

jesuarva committed:
SERVER: Implement starting Setup.

jesuarva committed:
ROUTERS: Define Starting files.

jesuarva committed:
WIRE server.js with Routers

jesuarva committed:
NOTES ROUTER && MODEL: Several actions.

1. ROUTER: Create endpoints.
2. MODEL: Define starting file with Mock data.
3. Test RouterFactory -> success.
4. Test connection to 'mongod' -> success.

jesuarva committed:
RouterFActory -> Several actions

1. Remove Folder
2. Add Touter factory via as a npm-dependency

jesuarva committed:
USERS ROUTER && MODEL: Several actions.

1. ROUTER: Create endpoints via express-router-factroy (This is module is mine ^_^).
2. MODEL: Define starting file.

jesuarva committed:
MODELS: Implement models for Notes and Users.

jesuarva committed:
HASH User password.

jesuarva committed:
FIX TYPO

jesuarva committed:
ENPOINT REGISTER: Implement register with JWT.

jesuarva committed:
FIX TYPO

Horribilous, I lost a lot of time because of this typo  :/

jesuarva committed:
ENPOINT LOGIN : Implement login with JWT.

jesuarva committed:
DATABASE: Connect to MLAB

jesuarva committed:
package.json -> remove 'start' script.

I am facing problems deploying to Heroku, even though I am no familiar
with npm Errors I noticed in the errors log a mention to nodemon and
the script, thus I am deleting that script aim to resolve the deployment
Erros.

jesuarva committed:
package.json -> Add 'strat' script. modify the value of the script:

Previous start script: "nodemon server.js".
New start script: "index.js"

jesuarva committed:
HEROKU ISSUES: Modify 'start' script in package.json

jesuarva committed:
HEROKU ISSUES: modify package.json

Pass 'dotenv' form dev-dependencies to dependencies.

jesuarva committed:
HEROKUISSUES: trying to circunvent 'dotenv' issue.

jesuarva committed:
HEROKU: Fix issue with dotenv.

NEW STATE:
1. Keep 'dotenv' as a dev-dependencies.
2. Add to index.js: if we are in NODE_ENV='production' do not call require('dotenv').config()

jesuarva committed:
AUTHENTICATION: Control access to Notes only to registered/loggedin users.

jesuarva committed:
NOTES: send only notes owned by the loggedin user.

jesuarva committed:
USERS: Set projections, thus the password is no send to the client APP.

nphillips78 commented on auth/loign.router.js:
```diff
@@ -0,0 +1,40 @@
+const express = require('express');
+const utils = require('./utils');
+const Users = require('../models/Users.model');
+const { RouterFactory } = require('express-router-factory');
```
Aren't you glad you took the time on the front end project to build some reusable elements?

jesuarva replied:
yes, very happy.
Did you noticed that I published it to npm? I use it as any other regular package dependency.
https://www.npmjs.com/package/express-router-factory

I still need to add a readme and instructions to the package in npm.

nphillips78 commented on auth/utils.js:
```diff
@@ -0,0 +1,21 @@
+      err ? res.status(500).json({ Error: 'Oh ohh, We can not validate your credentials, try again!' }) : next();
+    });
+  },
+};
```
I like this!

nphillips78 commented on db/db.js:
```diff
@@ -0,0 +1,13 @@
+  disconnect: function() {
+    return mongoose.disconnect();
+  },
+};
```
Beautiful

nphillips78 reviewed (commented):
Looks great so far! Let me know if you have any issues.

jesuarva committed:
APP INTEGRATION: disable control access to notes while setting up
the Frontend connection to API.

jesuarva committed:
APP INTEGRATION: disable control access-GET/notes whilst connecting to API.

jesuarva committed:
APP INTEGRATION: Modify Notes Model to fix APP strucutre.

Change APP struture is harder and error prone than chage only a value
here in the API's Notes Model.

jesuarva committed:
Playing with Passport.js

jesuarva committed:
DISALLOW Passport.js
